From f4777381ce10a5db904d120f4e14993af3dec39d Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@digia.com>
Date: Mon, 4 Mar 2013 16:32:02 +0100
Subject: [PATCH 1/2] [Qt] Flash objects won't load until scrolling page
 https://bugs.webkit.org/show_bug.cgi?id=110149

Reviewed by Simon Hausmann.

Revert r134222. The issue from bug 101836 is no longer reproducable,
and the fix was causing flash objects to not load properly.

* plugins/PluginPackage.cpp:
(WebCore::PluginPackage::determineQuirks):

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@144191 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 Source/WebCore/ChangeLog                 | 13 +++++++++++++
 Source/WebCore/plugins/PluginPackage.cpp |  4 +---
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/Source/WebCore/ChangeLog b/Source/WebCore/ChangeLog
index 833470c..a5acc4a 100644
--- a/Source/WebCore/ChangeLog
+++ b/Source/WebCore/ChangeLog
@@ -13,6 +13,19 @@
 
 2013-02-27  Allan Sandfeld Jensen  <allan.jensen@digia.com>
 
+        [Qt] Flash objects won't load until scrolling page
+        https://bugs.webkit.org/show_bug.cgi?id=110149
+
+        Reviewed by Simon Hausmann.
+
+        Revert r134222. The issue from bug 101836 is no longer reproducable,
+        and the fix was causing flash objects to not load properly.
+
+        * plugins/PluginPackage.cpp:
+        (WebCore::PluginPackage::determineQuirks):
+
+2013-02-27  Allan Sandfeld Jensen  <allan.jensen@digia.com>
+
         [TexMap] Flickering after transitions on Apple HTML5 demo
         https://bugs.webkit.org/show_bug.cgi?id=102501
 
diff --git a/Source/WebCore/plugins/PluginPackage.cpp b/Source/WebCore/plugins/PluginPackage.cpp
index 05b3eb6..5927719 100644
--- a/Source/WebCore/plugins/PluginPackage.cpp
+++ b/Source/WebCore/plugins/PluginPackage.cpp
@@ -205,9 +205,7 @@ void PluginPackage::determineQuirks(const String& mimeType)
         }
 
 #if PLATFORM(QT)
-        // Flash will crash on repeated calls to SetWindow in windowed mode.
-        // Defer the setWindow, so we don't set it to the wrong size too early.
-        m_quirks.add(PluginQuirkDeferFirstSetWindowCall);
+        // Flash will crash on repeated calls to SetWindow in windowed mode
         m_quirks.add(PluginQuirkDontCallSetWindowMoreThanOnce);
 #endif
 
-- 
1.8.1.4

